R/map2.R

```diff
@@ -60,9 +60,10 @@ map2_ <- function(.type,
                   .f,
                   ...,
                   .progress = FALSE,
+                  .purrr_user_env = caller_env(2),
                   .purrr_error_call = caller_env()) {
-  .x <- vctrs_vec_compat(.x)
-  .y <- vctrs_vec_compat(.y)
+  .x <- vctrs_vec_compat(.x, .purrr_user_env)
+  .y <- vctrs_vec_compat(.y, .purrr_user_env)
```

Comment:
If you do make `user_env` an optional arg here by giving it a default, I would then lean towards naming it explicitly here